koku/koku/trino_database.py

```diff
@@ -149,6 +148,15 @@ def executescript(trino_conn, sqlscript, *, params=None, preprocessor=None):
                 cur.execute(stmt, params=s_params)
                 results = cur.fetchall()
             except TrinoQueryError as trino_exc:
+                if "NoSuchKey" in str(trino_exc) and trino_external_error_retries > 0:
```

Comment:
Rather than using the `str` of the exception object, use [one of the properties](https://github.com/trinodb/trino-python-client/blob/master/trino/exceptions.py#L83-L114) for a more exact comparison.